Sources/SwiftGRPCNIO/CallHandlers/BidirectionalStreamingCallHandler.swift

```diff
@@ -21,14 +21,23 @@ public class BidirectionalStreamingCallHandler<RequestMessage: Message, Response
     self.context = context
     let eventObserver = eventObserverFactory(context)
     self.eventObserver = eventObserver
-    // Terminate the call if no observer is provided.
-    eventObserver.cascadeFailure(promise: context.statusPromise)
     context.statusPromise.futureResult.whenComplete {
       // When done, reset references to avoid retain cycles.
       self.eventObserver = nil
       self.context = nil
     }
   }
+  
+  public override func handlerAdded(ctx: ChannelHandlerContext) {
+    guard let eventObserver = eventObserver,
+      let context = context else { return }
+    // Terminate the call if providing an observer fails.
```

Comment:
I'm not sure I follow this line, do you mean "Terminate the call if the provided observer fails"?

The below explanation is helpful though! 👍

Reply:
Added "the future [providing an observer]", hope that makes it more clear: Users return a future containing the observer block. This lets them simply fail that future (with a `GRPCStatus`) if they do not want to receive messages on this call at all, e.g. in case of authentication failure (e.g. if the request metadata doesn't contain user credentials).